src/convertToObject.js

```diff
@@ -7,6 +7,26 @@
  */
 function convertToObject(sourceString) {
   // write your code here
+
+  const objConvert = {};
+
+  if (!sourceString.trim()) {
+    return objConvert;
+  }
+
+  sourceString.split(';').forEach((string) => {
+    const [key, value] = string
+      .split(':')
+      .map((item) => (item ? item.trim() : null));
+
+    if (key && value) {
+      const clearValue = value.replace(/;$/, '').trim();
```

Comment:
The line attempts to remove a trailing semicolon from the value. If the input format is strictly 'key:value', this might not be necessary. Ensure this behavior aligns with the intended input format.